imap_processing/cdf/cdf_attribute_manager.py

```diff
@@ -0,0 +1,207 @@
+        """
+        Load the default variable schema from the source directory.
+
+        Returns
+        -------
+        the dict representing the variable schema
+        """
+        default_schema_path = (
+            self.source_dir / "shared" / DEFAULT_VARIABLE_CDF_ATTRS_SCHEMA_FILE
+        )
+        # Load the Schema
+        return CdfAttributeManager._load_yaml_data(default_schema_path)
+
+    def load_global_attributes(self, file_path: str):
+        """
+        Update the global attributes property with the attributes from the file.
+
+        Parameters
+        ----------
+        file_path: str
+            file path to load, under self.source_dir.
+
+        """
+        self.global_attributes.update(self._load_yaml_data(self.source_dir / file_path))
+
+    @staticmethod
+    def _load_yaml_data(file_path: str | Path) -> dict:
+        """
+        Load a yaml file from the provided path.
+
+        Parameters
+        ----------
+        file_path: str | Path
+            path to the yaml file to load.
+
+        Returns
+        -------
+            Loaded yaml.
+        """
+        with open(file_path) as file:
+            return yaml.safe_load(file)
+
+    def get_global_attributes(self, instrument_id=None) -> dict:
+        """
+        Generate a dictionary global attributes based off the loaded schema and attrs.
+
+        Validates against the global schema to ensure all required variables are
+        present. It can also include instrument specific global attributes if
+        instrumet_id is set.
+
+        If a logical_source_id is provided, the level and instrument specific
+        attributes that were previously loaded using add_instrument_global_attrs will
+        be included.
+
+        Parameters
+        ----------
+        instrument_id : str
+            The id of the CDF file, used to retrieve instrument and level
+            specific global attributes. Suggested value is logical_source_id.
+
+        Returns
+        -------
+        output : dict
+            The global attribute values created from the input global attribute files
+            and schemas.
+        """
+        output = dict()
+        for attr_name, attr_schema in self.global_attribute_schema.items():
+            if attr_name in self.global_attributes:
+                output[attr_name] = self.global_attributes[attr_name]
+            # Retrieve instrument specific global attributes from the variable file
+            elif (
+                instrument_id is not None
+                and attr_name in self.global_attributes[instrument_id]
+            ):
+                output[attr_name] = self.global_attributes[instrument_id][attr_name]
+            elif attr_schema["required"] and attr_name not in self.global_attributes:
+                # TODO throw an error
+                output[attr_name] = None
+
+        return output
+
+    def load_variable_attrs(self, file_name: str) -> None:
```

Comment:
I think we should spell it out and be consistent with `load_global_attributes()`
```suggestion
    def load_variable_attributes(self, file_name: str) -> None:
```